packages/runtime/src/useCases/transport/relationshipTemplates/GetRelationshipTemplates.ts

```diff
@@ -51,6 +55,23 @@ export class GetRelationshipTemplatesUseCase extends UseCase<GetRelationshipTemp
                 (t) => t.maxNumberOfAllocations
             )}`,
             [nameof<RelationshipTemplateDTO>((r) => r.forIdentity)]: `${nameof<RelationshipTemplate>((r) => r.cache)}.${nameof<CachedRelationshipTemplate>((t) => t.forIdentity)}`
+        },
+        custom: {
+            [`${nameof<RelationshipTemplateDTO>((r) => r.passwordProtection)}.password`]: (query: any, input: string) => {
+                query[`${nameof<RelationshipTemplate>((t) => t.passwordProtection)}.${nameof<PasswordProtection>((t) => t.password)}`] = input;
+            },
+            [`${nameof<RelationshipTemplateDTO>((t) => t.passwordProtection)}.passwordIsPin`]: (query: any, input: string) => {
+                if (input === "true") {
+                    query[`${nameof<RelationshipTemplate>((t) => t.passwordProtection)}.${nameof<PasswordProtection>((t) => t.passwordType)}`] = {
+                        $regex: "^pin"
+                    };
+                }
+                if (input === "!") {
+                    query[`${nameof<RelationshipTemplate>((t) => t.passwordProtection)}.${nameof<PasswordProtection>((t) => t.passwordType)}`] = {
+                        $in: ["pw", undefined]
```

Comment:
```suggestion
                        $in: ["pw", undefined]
```

why do you want to say with `$in: undefined`? The field is never empty, right? So we should match `"pw"` exactly

Reply:
Technically speaking, passwordIsPin is undefined if there is no password protection

Reply:
But shouldn't this be matched in the password protection itself?

Reply:
I don't quite understand this question - do you assume that users will think that querying `passwordIsPin` will automatically restrict to password protected objects?

Reply:
IMHO, someone should query `"passwordProtection": "!"` when they want to find unprotected templates. It's very unintuitive to query `"passwordProtection.passwordIsPin": "!"` when wanting unprotected PWs.

Reply:
They can use `"passwordProtection.password": "!"` for unprotected templates

Reply:
the query for pw exactly here please

Reply:
`$in: ["pw", undefined]` is not a correct db syntax for all of our dbs, you would have to write:

```
$or: [
  {[`${nameof<RelationshipTemplate>((t) => t.passwordProtection)}.${nameof<PasswordProtection>((t) => **t.passwordType)}`]: { $exists: false }},
  {[`${nameof<RelationshipTemplate>((t) => t.passwordProtection)}.${nameof<PasswordProtection>((t) => **t.passwordType)}`]: "pw"},
]
```